From ac2c2ae95ab0fbbf75fd0a84fa33b5d1e6bb3782 Mon Sep 17 00:00:00 2001
From: Pzqqt <821026875@qq.com>
Date: Sun, 7 Apr 2024 19:51:08 +0800
Subject: [PATCH] crypto: zstd: Simplify `set_compression_level`

---
 crypto/zstd.c | 26 +++++++++++---------------
 1 file changed, 11 insertions(+), 15 deletions(-)

diff --git a/crypto/zstd.c b/crypto/zstd.c
index 1ec252c5196aa..cea47194a9e89 100644
--- a/crypto/zstd.c
+++ b/crypto/zstd.c
@@ -17,27 +17,23 @@
 
 static uint __read_mostly compression_level = 3;
 
-int set_compression_level(const char *val, const struct kernel_param *kp)
+static int set_compression_level(const char *buf, const struct kernel_param *kp)
 {
-	int temp, ret;
-
-	ret = sscanf(val, "%i", &temp);
-	if (ret == -EINVAL) {
-		return -EINVAL;
-	}
+	int ret;
+	uint temp;
 
-	if (temp == 0) {
-		temp = 1;
-	} else if (temp > zstd_max_clevel()) {
-		temp = zstd_max_clevel();
-	}
+	ret = kstrtouint(buf, 0, &temp);
+	if (ret)
+		return ret;
 
-	*((int *)kp->arg) = temp;
+	if (temp == 0 || temp > zstd_max_clevel())
+		return -EINVAL;
 
-    return 0;
+	return param_set_int(buf, kp);
 }
 
-module_param_call(compression_level, set_compression_level, param_get_int, &compression_level, 0644);
+module_param_call(compression_level, set_compression_level, param_get_int,
+		  &compression_level, 0644);
 
 struct zstd_ctx {
 	zstd_cctx *cctx;
