From 5e0c3ea38549209b2e60832c3b5bb797dccf7e11 Mon Sep 17 00:00:00 2001
From: hfdem <admin@hfdem.com>
Date: Tue, 26 Nov 2024 23:08:18 +0800
Subject: [PATCH] arm64: Optimize compiler for cortex-a510

---
 arch/arm64/Makefile | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 1384d7feec8b..894af83edd29 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -37,7 +37,11 @@ $(warning Detected assembler with broken .inst; disassembly will be unreliable)
 endif
 
 ifeq ($(shell test $(CONFIG_CLANG_VERSION) -gt 130000 2>/dev/null; echo $$?),0)
-KBUILD_CFLAGS	+= -mcpu=cortex-a510
+KBUILD_CFLAGS	+= -march=armv9-a+crypto+nosve -mcpu=cortex-a510
+KBUILD_AFLAGS   += -march=armv9-a+crypto+nosve -mcpu=cortex-a510
+endif
+ifeq ($(CONFIG_LD_IS_LLD), y)
+KBUILD_LDFLAGS  += -mllvm -march=armv9-a+crypto+nosve -mcpu=cortex-a510
 endif
 
 KBUILD_CFLAGS	+= -mgeneral-regs-only	\
